<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="escheR">
<title>Beyond Sequence-based Spatially-Resolved Data â€¢ escheR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Beyond Sequence-based Spatially-Resolved Data">
<meta property="og:description" content="escheR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">escheR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/SRT_eg.html">Getting Started with `escheR`</a>
    <a class="dropdown-item" href="../articles/more_than_visium.html">Beyond Sequence-based Spatially-Resolved Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/boyiguo1/escheR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Beyond Sequence-based Spatially-Resolved Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/boyiguo1/escheR/blob/HEAD/vignettes/more_than_visium.Rmd" class="external-link"><code>vignettes/more_than_visium.Rmd</code></a></small>
      <div class="d-none name"><code>more_than_visium.Rmd</code></div>
    </div>

    
    
<p>Starting from Version 1.2.0, <code>escheR</code> package supports
additional two data structures as input, including <a href="https://bioconductor.org/packages/release/bioc/html/SpatialExperiment.html" class="external-link"><code>SpatialExperiment</code></a>
and <code>data.frame</code> from <code>base</code> R. In addition,
<code>escheR</code> supports in-situ visualization of image-based
spatially resolved data, which will be the focus of future
development.</p>
<div class="section level2">
<h2 id="visualized-dimensionality-reduced-embedding-with-singlecellexperiment">Visualized Dimensionality Reduced Embedding with
<code>SingleCellExperiment</code><a class="anchor" aria-label="anchor" href="#visualized-dimensionality-reduced-embedding-with-singlecellexperiment"></a>
</h2>
<div class="section level3">
<h3 id="spatialexperiment-inherits-singlecellexperiment">
<code>SpatialExperiment</code> inherits
<code>SingleCellExperiment</code><a class="anchor" aria-label="anchor" href="#spatialexperiment-inherits-singlecellexperiment"></a>
</h3>
<p>Following the same syntax, one can also visualize dimensionality
reduced embeddings of a <code>SpatialExperiment</code> object by
providing the argument <code>dimred</code> with a non-null value. Hence,
the first 2 columns of the corresponding <code>reducedDim(spe)</code>
assay will be used as the x-y coordinate of the plot, replacing
<code>spatialCoords(spe)</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boyiguo1/escheR" class="external-link">escheR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/STexampleData" class="external-link">STexampleData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/STexampleData.html" class="external-link">Visium_humanDLPFC</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="va">spe</span><span class="op">$</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">ground_truth</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">top.gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html" class="external-link">getTopHVGs</a></span><span class="op">(</span><span class="va">spe</span>, n<span class="op">=</span><span class="fl">500</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span> <span class="co"># See below.</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">spe</span>, subset_row <span class="op">=</span> <span class="va">top.gene</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span></span>
<span>  <span class="va">spe</span>,</span>
<span>  dimred <span class="op">=</span> <span class="st">"PCA"</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_fill.html">add_fill</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"ground_truth"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="more_than_visium_files/figure-html/redDim_spe-1.png"><!-- --></p>
</div>
</div>
<div class="section level2">
<h2 id="hex-binning">Hex Binning<a class="anchor" aria-label="anchor" href="#hex-binning"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span><span class="op">$</span><span class="va">counts_MOBP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span><span class="op">==</span><span class="st">"MOBP"</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">ground_truth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">ground_truth</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Point Binning version</span></span>
<span><span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span></span>
<span>  <span class="va">spe</span>,</span>
<span>  dimred <span class="op">=</span> <span class="st">"PCA"</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_ground.html">add_ground_bin</a></span><span class="op">(</span></span>
<span>    var <span class="op">=</span> <span class="st">"ground_truth"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_fill.html">add_fill_bin</a></span><span class="op">(</span></span>
<span>    var <span class="op">=</span> <span class="st">"counts_MOBP"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># Customize aesthetics</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"white"</span>, high <span class="op">=</span> <span class="st">"black"</span>, name <span class="op">=</span> <span class="st">"MOBP Count"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_color_discrete</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Spatial Domains"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="more_than_visium_files/figure-html/bin_plot-1.png"><!-- -->
&gt; Note 1: The strategy of binning to avoid overplotting is previously
proposed in <a href="https://www.bioconductor.org/packages/release/bioc/html/schex.html" class="external-link"><code>schex</code></a>.
While we provide an implementation in <code>escheR</code>, we would
caution our users that the binning strategy could lead to intermixing of
cluster memberships. In our implementation, the majority membership of
the data points belonging to a bin is selected as the label of the bin.
Users should use the binning strategy under their own discretion, and
interpret the visualization carefully.</p>
<blockquote>
<p>Note 2: <code><a href="../reference/add_fill.html">add_fill_bin()</a></code> shoudl be applied after
<code><a href="../reference/add_ground.html">add_ground_bin()</a></code> for the better visualization outcome.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="image-based-spatialexperiment-object">Image-based <code>SpatialExperiment</code> Object<a class="anchor" aria-label="anchor" href="#image-based-spatialexperiment-object"></a>
</h2>
<p>To demonstrate the principle that <code>escheR</code> can be used to
visualize image-based spatially-resolved data pending optimization, we
include two image-based spatially resolved transcriptomics data
generated via seqFish platform and Slide-seq V2 platform respectively.
The two datasets have been previously curated in the <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/STexampleData/inst/doc/STexampleData_overview.html" class="external-link"><code>STexampleData</code></a>
package</p>
<div class="section level3">
<h3 id="seqfish">seqFISH<a class="anchor" aria-label="anchor" href="#seqfish"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/STexampleData" class="external-link">STexampleData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boyiguo1/escheR" class="external-link">escheR</a></span><span class="op">)</span></span>
<span><span class="va">spe_seqFISH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/STexampleData.html" class="external-link">seqFISH_mouseEmbryo</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span><span class="va">spe_seqFISH</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/add_fill.html">add_fill</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"embryo"</span><span class="op">)</span></span></code></pre></div>
<p><img src="more_than_visium_files/figure-html/im_seqFISH-1.png"><!-- --></p>
<blockquote>
<p>NOTE: trimming down the <code>colData(spe)</code> before piping into
make-escheR could reduce the computation time to make the plots,
specifically when <code>colData(spe)</code> contains extremely large
number of irrelavent features/columns.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="slideseqv2">SlideSeqV2<a class="anchor" aria-label="anchor" href="#slideseqv2"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/STexampleData" class="external-link">STexampleData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boyiguo1/escheR" class="external-link">escheR</a></span><span class="op">)</span></span>
<span><span class="va">spe_slideseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/STexampleData.html" class="external-link">SlideSeqV2_mouseHPC</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span><span class="va">spe_slideseq</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/add_fill.html">add_fill</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"celltype"</span><span class="op">)</span></span></code></pre></div>
<p><img src="more_than_visium_files/figure-html/im_slideseq-1.png"><!-- --></p>
</div>
</div>
<div class="section level2">
<h2 id="beyond-bioconductor-eco-system">Beyond Bioconductor Eco-system<a class="anchor" aria-label="anchor" href="#beyond-bioconductor-eco-system"></a>
</h2>
<p>We aim to provide accessibility to all users regardless of their
programming background and preferred single-cell analysis pipelines.
Nevertheless , with limited resource, our sustaining efforts will
prioritize towards the maintenance of the established functionality and
the optimization for image-based spatially resolved data. We regret we
are not be able to provide seamless interface to other R pipelines such
as <code>Seurat</code> and <code>Giotto</code> in foreseeable
future.</p>
<p>Instead, we provide a generic function that works with a
<code>data.frame</code> object as input. For example, relevant features
in <code>Suerat</code> can be easily exported as a
<code>data.frame</code> object manually or via
<code>tidyseurat</code>[<a href="https://github.com/stemangiola/tidyseurat" class="external-link uri">https://github.com/stemangiola/tidyseurat</a>]. The exported
data frame can be pipe into <code>escheR</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boyiguo1/escheR" class="external-link">escheR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="va">pbmc_small</span> <span class="op">&lt;-</span> <span class="fu">SeuratObject</span><span class="fu">::</span><span class="va"><a href="https://mojaveazure.github.io/seurat-object/reference/pbmc_small.html" class="external-link">pbmc_small</a></span></span>
<span><span class="va">pbmc_2pc</span> <span class="op">&lt;-</span> <span class="va">pbmc_small</span><span class="op">@</span><span class="va">reductions</span><span class="op">$</span><span class="va">pca</span><span class="op">@</span><span class="va">cell.embeddings</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">pbmc_meta</span> <span class="op">&lt;-</span> <span class="va">pbmc_small</span><span class="op">@</span><span class="va">meta.data</span></span>
<span></span>
<span><span class="co">#&gt; Call generic function for make_escheR.data.frame</span></span>
<span><span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">pbmc_meta</span>,</span>
<span>  .x <span class="op">=</span> <span class="va">pbmc_2pc</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  .y <span class="op">=</span> <span class="va">pbmc_2pc</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_fill.html">add_fill</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"groups"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] BumpyMatrix_1.10.0          scran_1.30.0               </span></span>
<span><span class="co">#&gt;  [3] scater_1.30.0               scuttle_1.12.0             </span></span>
<span><span class="co">#&gt;  [5] STexampleData_1.10.0        SpatialExperiment_1.12.0   </span></span>
<span><span class="co">#&gt;  [7] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0</span></span>
<span><span class="co">#&gt;  [9] Biobase_2.62.0              GenomicRanges_1.54.1       </span></span>
<span><span class="co">#&gt; [11] GenomeInfoDb_1.38.0         IRanges_2.36.0             </span></span>
<span><span class="co">#&gt; [13] S4Vectors_0.40.1            MatrixGenerics_1.14.0      </span></span>
<span><span class="co">#&gt; [15] matrixStats_1.1.0           ExperimentHub_2.10.0       </span></span>
<span><span class="co">#&gt; [17] AnnotationHub_3.10.0        BiocFileCache_2.10.1       </span></span>
<span><span class="co">#&gt; [19] dbplyr_2.4.0                BiocGenerics_0.48.1        </span></span>
<span><span class="co">#&gt; [21] escheR_1.3.1                ggplot2_3.4.4              </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_1.8.7                magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;   [3] ggbeeswarm_0.7.2              magick_2.8.1                 </span></span>
<span><span class="co">#&gt;   [5] farver_2.1.1                  rmarkdown_2.25               </span></span>
<span><span class="co">#&gt;   [7] fs_1.6.3                      zlibbioc_1.48.0              </span></span>
<span><span class="co">#&gt;   [9] ragg_1.2.6                    vctrs_0.6.4                  </span></span>
<span><span class="co">#&gt;  [11] memoise_2.0.1                 DelayedMatrixStats_1.24.0    </span></span>
<span><span class="co">#&gt;  [13] RCurl_1.98-1.13               htmltools_0.5.7              </span></span>
<span><span class="co">#&gt;  [15] S4Arrays_1.2.0                curl_5.1.0                   </span></span>
<span><span class="co">#&gt;  [17] BiocNeighbors_1.20.0          SparseArray_1.2.2            </span></span>
<span><span class="co">#&gt;  [19] sass_0.4.7                    bslib_0.5.1                  </span></span>
<span><span class="co">#&gt;  [21] desc_1.4.2                    cachem_1.0.8                 </span></span>
<span><span class="co">#&gt;  [23] igraph_1.5.1                  mime_0.12                    </span></span>
<span><span class="co">#&gt;  [25] lifecycle_1.0.4               pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt;  [27] rsvd_1.0.5                    Matrix_1.6-2                 </span></span>
<span><span class="co">#&gt;  [29] R6_2.5.1                      fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [31] GenomeInfoDbData_1.2.11       shiny_1.7.5.1                </span></span>
<span><span class="co">#&gt;  [33] digest_0.6.33                 colorspace_2.1-0             </span></span>
<span><span class="co">#&gt;  [35] AnnotationDbi_1.64.1          rprojroot_2.0.4              </span></span>
<span><span class="co">#&gt;  [37] dqrng_0.3.1                   irlba_2.3.5.1                </span></span>
<span><span class="co">#&gt;  [39] textshaping_0.3.7             RSQLite_2.3.3                </span></span>
<span><span class="co">#&gt;  [41] beachmat_2.18.0               labeling_0.4.3               </span></span>
<span><span class="co">#&gt;  [43] filelock_1.0.2                fansi_1.0.5                  </span></span>
<span><span class="co">#&gt;  [45] httr_1.4.7                    abind_1.4-5                  </span></span>
<span><span class="co">#&gt;  [47] compiler_4.3.2                bit64_4.0.5                  </span></span>
<span><span class="co">#&gt;  [49] withr_2.5.2                   BiocParallel_1.36.0          </span></span>
<span><span class="co">#&gt;  [51] viridis_0.6.4                 DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [53] hexbin_1.28.3                 highr_0.10                   </span></span>
<span><span class="co">#&gt;  [55] rappdirs_0.3.3                DelayedArray_0.28.0          </span></span>
<span><span class="co">#&gt;  [57] rjson_0.2.21                  bluster_1.12.0               </span></span>
<span><span class="co">#&gt;  [59] tools_4.3.2                   vipor_0.4.5                  </span></span>
<span><span class="co">#&gt;  [61] beeswarm_0.4.0                interactiveDisplayBase_1.40.0</span></span>
<span><span class="co">#&gt;  [63] httpuv_1.6.12                 glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [65] promises_1.2.1                grid_4.3.2                   </span></span>
<span><span class="co">#&gt;  [67] cluster_2.1.4                 generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [69] gtable_0.3.4                  metapod_1.10.0               </span></span>
<span><span class="co">#&gt;  [71] BiocSingular_1.18.0           ScaledMatrix_1.10.0          </span></span>
<span><span class="co">#&gt;  [73] utf8_1.2.4                    XVector_0.42.0               </span></span>
<span><span class="co">#&gt;  [75] ggrepel_0.9.4                 BiocVersion_3.18.0           </span></span>
<span><span class="co">#&gt;  [77] pillar_1.9.0                  stringr_1.5.0                </span></span>
<span><span class="co">#&gt;  [79] limma_3.58.1                  later_1.3.1                  </span></span>
<span><span class="co">#&gt;  [81] dplyr_1.1.3                   lattice_0.22-5               </span></span>
<span><span class="co">#&gt;  [83] bit_4.0.5                     tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt;  [85] locfit_1.5-9.8                Biostrings_2.70.1            </span></span>
<span><span class="co">#&gt;  [87] knitr_1.45                    gridExtra_2.3                </span></span>
<span><span class="co">#&gt;  [89] edgeR_4.0.1                   xfun_0.41                    </span></span>
<span><span class="co">#&gt;  [91] statmod_1.5.0                 stringi_1.7.12               </span></span>
<span><span class="co">#&gt;  [93] yaml_2.3.7                    evaluate_0.23                </span></span>
<span><span class="co">#&gt;  [95] codetools_0.2-19              tibble_3.2.1                 </span></span>
<span><span class="co">#&gt;  [97] BiocManager_1.30.22           cli_3.6.1                    </span></span>
<span><span class="co">#&gt;  [99] xtable_1.8-4                  systemfonts_1.0.5            </span></span>
<span><span class="co">#&gt; [101] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt; [103] Rcpp_1.0.11                   png_0.1-8                    </span></span>
<span><span class="co">#&gt; [105] parallel_4.3.2                ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt; [107] pkgdown_2.0.7                 blob_1.2.4                   </span></span>
<span><span class="co">#&gt; [109] sparseMatrixStats_1.14.0      bitops_1.0-7                 </span></span>
<span><span class="co">#&gt; [111] viridisLite_0.4.2             scales_1.2.1                 </span></span>
<span><span class="co">#&gt; [113] purrr_1.0.2                   crayon_1.5.2                 </span></span>
<span><span class="co">#&gt; [115] rlang_1.1.2                   KEGGREST_1.42.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Boyi Guo, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
