<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="escheR">
<title>Getting Started with `escheR` • escheR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with `escheR`">
<meta property="og:description" content="escheR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">escheR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/SRT_eg.html">Getting Started with `escheR`</a>
    <a class="dropdown-item" href="../articles/more_than_visium.html">Beyond Sequence-based Spatially-Resolved Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/boyiguo1/escheR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Getting Started with `escheR`</h1>
                        <h4 data-toc-skip class="author">Boyi Guo</h4>
            <address class="author_afil">
      Johns Hopkins Bloomberg School of Public Health, Baltimore, MD,
USA<br><h4 data-toc-skip class="author">Stephanie
C. Hicks</h4>
            <address class="author_afil">
      Johns Hopkins Bloomberg School of Public Health, Baltimore, MD,
USA<br><small class="dont-index">Source: <a href="https://github.com/boyiguo1/escheR/blob/HEAD/vignettes/SRT_eg.Rmd" class="external-link"><code>vignettes/SRT_eg.Rmd</code></a></small>
      <div class="d-none name"><code>SRT_eg.Rmd</code></div>
    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The goal of <code>escheR</code> is to create an unified
multi-dimensional spatial visualizations for spatially-resolved
transcriptomics data following Gestalt principles.</p>
<p>Our preprint describing the innovative visualization is available via
<a href="https://www.biorxiv.org/content/10.1101/2023.03.18.533302" class="external-link">bioRxiv</a>.</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>You can install the latest release version of <code>escheR</code>
from Bioconductor via the following code. Additional details are shown
on the <a href="https://bioconductor.org/packages/escheR" class="external-link">Bioconductor</a>
page.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"escheR"</span><span class="op">)</span></span></code></pre></div>
<p>The latest development version can also be installed from the
<code>devel</code> version of Bioconductor or from <a href="https://github.com/boyiguo1/escheR" class="external-link">GitHub</a> following</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"boyiguo1/escheR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># `devel` version from Bioconductor</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span>version<span class="op">=</span><span class="st">'devel'</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="input-data-format">Input data format<a class="anchor" aria-label="anchor" href="#input-data-format"></a>
</h2>
<p>Starting from Version 1.2.0, <code>escheR</code> package supports
three data structures, including <a href="https://bioconductor.org/packages/release/bioc/html/SpatialExperiment.html" class="external-link"><code>SpatialExperiment</code></a>,
<a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link"><code>SingleCellExperiment</code></a>,
and <code>data.frame</code> from <code>base</code> R.</p>
<p>In the following example, we demonstrate the how to use
<code>escheR</code> with a <a href="https://bioconductor.org/packages/release/bioc/html/SpatialExperiment.html" class="external-link"><code>SpatialExperiment</code></a>
object. Please visit our other tutorials for [TODO: add items and
list].</p>
</div>
<div class="section level2">
<h2 id="making-escher-plot">Making escheR Plot<a class="anchor" aria-label="anchor" href="#making-escher-plot"></a>
</h2>
<div class="section level3">
<h3 id="load-packages">Load Packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<p>To run the demonstration, there are two necessary packages to load,
<code>escheR</code> and <code>STexampleData</code>. <a href="https://bioconductor.org/packages/STexampleData" class="external-link"><code>STexampleData</code></a>
contains a pre-processed 10x Visium dataset.</p>
<p>To note, <code>escheR</code> will automatically load
<code>ggplot2</code> package. Hence, explicitly loading
<code>ggplot2</code> is not required.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boyiguo1/escheR" class="external-link">escheR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/STexampleData" class="external-link">STexampleData</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="preparing-example-data">Preparing example data<a class="anchor" aria-label="anchor" href="#preparing-example-data"></a>
</h3>
<p>In this step, we will find one 10xVisium sample from <a href="https://bioconductor.org/packages/release/data/experiment/html/STexampleData.html" class="external-link"><code>STexampleData</code></a>
package, indexed by brain number of “151673”. For more information,
please see the vignettes of <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/STexampleData/inst/doc/STexampleData_overview.html" class="external-link"><code>STexampleData</code></a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/STexampleData.html" class="external-link">Visium_humanDLPFC</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Subset in-tissue spots</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="va">spe</span><span class="op">$</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">ground_truth</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>Here is a summary of the <code>SpatialExperiment</code> object called
<code>spe</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span></span>
<span><span class="co">#&gt; class: SpatialExperiment </span></span>
<span><span class="co">#&gt; dim: 33538 3611 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span></span>
<span><span class="co">#&gt;   ENSG00000268674</span></span>
<span><span class="co">#&gt; rowData names(3): gene_id gene_name feature_type</span></span>
<span><span class="co">#&gt; colnames(3611): AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 ...</span></span>
<span><span class="co">#&gt;   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1</span></span>
<span><span class="co">#&gt; colData names(7): barcode_id sample_id ... ground_truth cell_count</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span><span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="set-up-an-escher-plot">Set up an <code>escheR</code> plot<a class="anchor" aria-label="anchor" href="#set-up-an-escher-plot"></a>
</h3>
<p>Similar to <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot2::ggplot()</a></code>, we first use the function
<code><a href="../reference/make_escheR.html">make_escheR()</a></code> to create an empty plot. The input of
<code><a href="../reference/make_escheR.html">make_escheR()</a></code> is a <a href="https://bioconductor.org/packages/release/bioc/html/SpatialExperiment.html" class="external-link"><code>SpatialExperiment</code></a>
object. The output of the function is a <code>ggplot</code> object with
no layer in it.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="adding-layers">Adding layers<a class="anchor" aria-label="anchor" href="#adding-layers"></a>
</h3>
<p>Unlike <code>ggplot2</code>, we use piping <code>|&gt;</code> instead
of <code>+</code> to apply layers the figure. Mainly, we have three
functions <code>add_fill</code>, <code>add_ground</code>,
<code>add_symbol</code>. The inputs of these <code>add_*</code>
functions include the plots created using <code>make_scheR()</code> and
the variable name for the layer. Currently, the variable name should be
in the the column data of the <code>spe</code> object,
i.e. <code>colData(spe)</code>.</p>
<p>Here we first apply the <code>add_fill</code> to add the spots
color-coded by the total number of cells all
spots(<code>sum_umi</code>).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">|&gt;</span></span>
<span>   <span class="fu"><a href="../reference/add_fill.html">add_fill</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"cell_count"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/creat_fill-1.png"><!-- --></p>
<p>It is okay to use any combination of the <code>add_*</code>
functions. For example, we want to show the spatial domains of the
samples as the ground of the figure and use symbols to denote if each
spot is within the outline of the tissue slice. In this example, all
plotted spots are in the outlines of the tissue slice and hence marked
with dots.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">|&gt;</span></span>
<span>   <span class="fu"><a href="../reference/add_ground.html">add_ground</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"ground_truth"</span><span class="op">)</span><span class="op">)</span> <span class="co"># round layer</span></span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/create_ground-1.png"><!-- --></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/add_symbol.html">add_symbol</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"ground_truth"</span>, size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="co"># Symbol layer</span></span>
<span><span class="co">#&gt; Warning: The shape palette can deal with a maximum of 6 discrete values because</span></span>
<span><span class="co">#&gt; more than 6 becomes difficult to discriminate; you have 7. Consider</span></span>
<span><span class="co">#&gt; specifying shapes manually if you must have them.</span></span>
<span><span class="co">#&gt; Warning: Removed 513 rows containing missing values (`geom_point()`).</span></span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/add_symbol-1.png"><!-- --></p>
<p>It is okay to change the ordering of these <code>add_*</code>
functions. However, we advise to always have the <code>add_fill</code>
as the first step to achieve the best visual effect due to the laying
mechanism.</p>
</div>
</div>
<div class="section level2">
<h2 id="customize-escher-plot">Customize escheR Plot<a class="anchor" aria-label="anchor" href="#customize-escher-plot"></a>
</h2>
<div class="section level3">
<h3 id="choosing-color-palette-for-add_fill-and-add_ground">Choosing Color Palette for <code>add_fill</code> and
<code>add_ground</code><a class="anchor" aria-label="anchor" href="#choosing-color-palette-for-add_fill-and-add_ground"></a>
</h3>
<p>To maximize the utility of the multi-dimensional plot by applying
both color-coded layers using <code><a href="../reference/add_fill.html">add_fill()</a></code> and
<code><a href="../reference/add_ground.html">add_ground()</a></code>, it is important to choose minimally
interfering color-palette for the <code>fill</code> and
<code>ground</code> to avoid visualization confusion. The following
demonstration provide some examples for simultaneously visualization of
two variables regardless of their types (continuous vs categorical, or
categorical vs categorical.)</p>
<div class="section level4">
<h4 id="coninuous-variable-gene-expression-vs-categorical-variable-spatial-domains">Coninuous variable (gene expression) vs Categorical variable
(Spatial Domains)<a class="anchor" aria-label="anchor" href="#coninuous-variable-gene-expression-vs-categorical-variable-spatial-domains"></a>
</h4>
<p>The following example visualizes the differential gene expression of
<em>MOBP</em>, a marker gene for white matter, across different spatial
domains. The default color palette, <code>viridis</code>, are not easily
visible with color-coded spatial domains as there are overlapping in the
color space, which could lead to possible confusion.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prep data</span></span>
<span><span class="co"># Adding gene counts for MBP to the colData</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">counts_MOBP</span> <span class="op">&lt;-</span> <span class="fu">counts</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span><span class="op">==</span><span class="st">"MOBP"</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="op">(</span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="../reference/add_fill.html">add_fill</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"counts_MOBP"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="../reference/add_ground.html">add_ground</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"ground_truth"</span>, stroke <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/default_con_cat_eg-1.png"><!-- --></p>
<p>To improve the visualization, we choose to use a color that is not
included in the color palette for <code>ground_truth</code>, which is
the color <em>black</em>. Specifically, we use a color gradient from
white (no expression) to black (maximum of gene counts) to represent the
expression of MOBP in each spot. By using the white-black color gradient
for the gene expression, we minimize the overlapping of the choice of
color for spatial domains.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> </span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"white"</span>, high <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/improved_con_cat_eg-1.png"><!-- --></p>
<p>After customizing the color palettes to be minimally overlapping, it
is easier to observe that <em>MOBP</em> has higher raw gene counts in
the white matter (<code>WM</code>) region than other regions.</p>
</div>
<div class="section level4">
<h4 id="categorical-variable-vs-categorical-variable">Categorical variable Vs Categorical variable<a class="anchor" aria-label="anchor" href="#categorical-variable-vs-categorical-variable"></a>
</h4>
<p>In this example, we demonstrate how to optimize color palettes for
visualizing two categorical variables. We first create an arbitrary
5-level categorical variable called <code>tmp_group</code>, representing
different horizontal regions of the tissue section.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span><span class="op">$</span><span class="va">tmp_group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cut.html" class="external-link">cut</a></span><span class="op">(</span></span>
<span>  <span class="va">spe</span><span class="op">$</span><span class="va">array_row</span>, </span>
<span>  breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">array_row</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span> ,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/stats/fivenum.html" class="external-link">fivenum</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">array_row</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">tmp_group</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  (-1,0]  (0,18] (18,36] (36,52] (52,73] </span></span>
<span><span class="co">#&gt;      40     864     933     893     881</span></span></code></pre></div>
<p>Following the principle to avoid overlapping of two color palettes,
we use gradients of blue for different levels of
<code>tmp_group</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_fill.html">add_fill</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"tmp_group"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_ground.html">add_ground</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"ground_truth"</span>, stroke <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_void</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/cat_cat_eg-1.png"><!-- --></p>
<p>Here is another example where we try another manually-curated
color-palette. We follow the same principle, minimize overlapping of two
color-palettes for ground (<code>scale_color_manual</code>) and fill
(<code>scale_fill_brewer</code>) respectively. Specifically, we use
gradients of blue to show <code>tmp_group</code> and other colors for
spatial domains <code>ground_truth</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_fill.html">add_fill</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"tmp_group"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_ground.html">add_ground</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"ground_truth"</span>, stroke <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">""</span>, <span class="co"># turn off legend name for ground_truth</span></span>
<span>    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      <span class="st">"Layer1"</span> <span class="op">=</span> <span class="st">"#F0027F"</span>,</span>
<span>      <span class="st">"Layer2"</span> <span class="op">=</span> <span class="st">"transparent"</span>,</span>
<span>      <span class="st">"Layer3"</span> <span class="op">=</span> <span class="st">"#4DAF4A"</span>,</span>
<span>      <span class="st">"Layer4"</span> <span class="op">=</span> <span class="st">"#984EA3"</span>,</span>
<span>      <span class="st">"Layer5"</span> <span class="op">=</span> <span class="st">"#FFD700"</span>,</span>
<span>      <span class="st">"Layer6"</span> <span class="op">=</span> <span class="st">"#FF7F00"</span>,</span>
<span>      <span class="st">"WM"</span> <span class="op">=</span> <span class="st">"#1A1A1A"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> </span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/cat_cat_eg2-1.png"><!-- --></p>
<blockquote>
<p>NOTE: It would be intrinsically difficult to construct a clear,
intuitive and precise visualization when the number of categories/levels
is large.</p>
</blockquote>
</div>
<div class="section level4">
<h4 id="guidance-in-choosing-bivariate-color-palette">Guidance in choosing bivariate color palette<a class="anchor" aria-label="anchor" href="#guidance-in-choosing-bivariate-color-palette"></a>
</h4>
<p>In this vignettes, we don’t provide or recommend specific color
palettes, because the selection of color palettes is highly relevant to
the underlying message and heterogeneous across analysis and studies,
e.g. sequential palettes, qualitative palette, and divergent palette.
Instead, we direct interested user to explore the topic on <em>bivariate
color palette</em>. The <a href="https://jakubnowosad.com/posts/2020-08-25-cbc-bp2/" class="external-link">blog post</a>
by Jakub Nowosad and R package <a href="https://cran.r-project.org/web/packages/biscale/vignettes/bivariate_palettes.html" class="external-link"><code>biscale</code></a>
could be helpful to optimize your color palette for bivariate
visualization.</p>
<p>In addition, if color palette is extremely to curate, e.g. large
number of levels, it is possible to use symbols (<a href="#id_#Adding-layers"><code>add_symbol()</code></a>) to annotate
specific levels to avoid clutter in the color space.</p>
</div>
</div>
<div class="section level3">
<h3 id="adjusting-aesthetics">Adjusting aesthetics<a class="anchor" aria-label="anchor" href="#adjusting-aesthetics"></a>
</h3>
<p>Given that the <code>escheR</code> package is developed based on
<code>ggplot2</code>, aesthetics can be easily adjusted following the
<code>ggplot2</code> syntax. For example, given a <code>escheR</code>
plot object, one can use <code>+</code> with <code>theme_*</code>,
<code>scale_*</code> functions.</p>
<p>For example, to change the aesthetics of each layer, one can simply
use the <code>scale_*</code> from <code>ggplot2</code> to optimize the
final visual presentation. For example, to optimize
<code>add_fill</code>, one can use <code>scale_fill_*</code>; to
optimize <code>add_ground</code>, one can use
<code>scale_color_*</code>; to optimize <code>add_sumbol</code>, one use
<code>scale_shape_*</code>. Here, we demonstrate how to change the color
for the ground layer ( <code>add_ground</code>) using
<code>scale_color_manual</code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">p_final</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">""</span>, <span class="co"># No legend name</span></span>
<span>    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      <span class="st">"Layer1"</span> <span class="op">=</span> <span class="st">"#F0027F"</span>,</span>
<span>      <span class="st">"Layer2"</span> <span class="op">=</span> <span class="st">"#377EB8"</span>,</span>
<span>      <span class="st">"Layer3"</span> <span class="op">=</span> <span class="st">"#4DAF4A"</span>,</span>
<span>      <span class="st">"Layer4"</span> <span class="op">=</span> <span class="st">"#984EA3"</span>,</span>
<span>      <span class="st">"Layer5"</span> <span class="op">=</span> <span class="st">"#FFD700"</span>,</span>
<span>      <span class="st">"Layer6"</span> <span class="op">=</span> <span class="st">"#FF7F00"</span>,</span>
<span>      <span class="st">"WM"</span> <span class="op">=</span> <span class="st">"#1A1A1A"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Example Title"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="show-a-subset-of-levels">Show a subset of levels<a class="anchor" aria-label="anchor" href="#show-a-subset-of-levels"></a>
</h3>
<p>The easiest way to show only a subset of levels of a categorical
variable is to create a new variable where all the unwanted levels will
be set to NA values. Please see the example below</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">ground_truth</span>, useNA <span class="op">=</span> <span class="st">"ifany"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Layer1 Layer2 Layer3 Layer4 Layer5 Layer6     WM </span></span>
<span><span class="co">#&gt;    273    253    989    218    673    692    513</span></span>
<span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">tmp_fac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">ground_truth</span>,</span>
<span>                      levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Layer1"</span>, <span class="st">"Layer2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">tmp_fac</span>, useNA <span class="op">=</span> <span class="st">"ifany"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Layer1 Layer2   &lt;NA&gt; </span></span>
<span><span class="co">#&gt;    273    253   3085</span></span>
<span></span>
<span><span class="fu"><a href="../reference/make_escheR.html">make_escheR</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_ground.html">add_ground</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"ground_truth"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/add_symbol.html">add_symbol</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"tmp_fac"</span>, size <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_shape_manual</a></span><span class="op">(</span></span>
<span>    values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">16</span><span class="op">)</span>,    <span class="co">#&gt; Set different symbols for the 2 levels</span></span>
<span>    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Layer1"</span>, <span class="st">"Layer2"</span><span class="op">)</span> <span class="co">#&gt; Remove NA from legend</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 3085 rows containing missing values (`geom_point()`).</span></span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/subset-1.png"><!-- --></p>
</div>
<div class="section level3">
<h3 id="multi-sample-figure">Multi-sample Figure<a class="anchor" aria-label="anchor" href="#multi-sample-figure"></a>
</h3>
<p>By design, <code>make_escheR</code> operates on only one sample. In
order to create a figure compiling the spatial plots for multiple
samples, individual plots are required via a series of calls to
<code>make_escheR</code>, possibly via a <code>for</code> loop or an
iterator function (e.g. <code>lapply</code>).</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># Create a list of `escheR` plots</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">unique</span>(spe<span class="sc">$</span>sample_id) <span class="sc">|&gt;</span> <span class="co"># Create a list of sample names</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>  <span class="fu">lapply</span>(<span class="at">FUN =</span> <span class="cf">function</span>(.sample_id){ <span class="co"># Iterate over all samples</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>    spe_single <span class="ot">&lt;-</span> spe[, spe<span class="sc">$</span>sample_id <span class="sc">==</span> .sample_id]</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>    <span class="fu">make_escheR</span>(spe_single) <span class="sc">|&gt;</span> </span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>      <span class="fu">add_fill</span>(<span class="at">var =</span> <span class="st">"counts_MOBP"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>      <span class="fu">add_ground</span>(<span class="at">var =</span> <span class="st">"ground_truth"</span>, <span class="at">stroke =</span> <span class="fl">0.5</span>)<span class="er">)</span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a><span class="co"># Customize theme</span></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>  })</span></code></pre></div>
<p>Given all plots made for individual samples are stored in a preferred
data structure (e.g. a <code>list</code>), one can use many functions,
e.g. <a href="https://wilkelab.org/cowplot/articles/plot_grid.html" class="external-link"><code>cowplot::plot_grid</code></a>,
<a href="https://patchwork.data-imaginist.com/index.html" class="external-link"><code>patchwork</code></a>,
to compile and arrange individual plots to a paneled figure. The
following example uses <a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link"><code>ggpubr::ggarrange</code></a>
to create a figure from a list of <code>escheR</code> plots.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span>
<span><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">p2</span>, <span class="va">p2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span></span>
<span>  plotlist <span class="op">=</span> <span class="va">plot_list</span>,</span>
<span>  ncol <span class="op">=</span> <span class="fl">2</span>, nrow <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  common.legend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="SRT_eg_files/figure-html/paneling-1.png"><!-- --></p>
</div>
</div>
<div class="section level2">
<h2 id="save-escher-plot">Save escheR plot<a class="anchor" aria-label="anchor" href="#save-escher-plot"></a>
</h2>
<p>The procedure to save <code>escheR</code> plots is exactly the same
as saving a <code>ggplot</code> object. In the example below, we use the
function <code><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggplot2::ggsave()</a></code> to save <code>escheR</code>
plots in the <code>pdf</code> format.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span></span>
<span>  filename <span class="op">=</span> <span class="st">"path/file_name.pdf"</span>,</span>
<span>  plot <span class="op">=</span> <span class="va">p_final</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggpubr_0.6.0                STexampleData_1.10.0       </span></span>
<span><span class="co">#&gt;  [3] SpatialExperiment_1.12.0    SingleCellExperiment_1.24.0</span></span>
<span><span class="co">#&gt;  [5] SummarizedExperiment_1.32.0 Biobase_2.62.0             </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.54.1        GenomeInfoDb_1.38.0        </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.36.0              S4Vectors_0.40.1           </span></span>
<span><span class="co">#&gt; [11] MatrixGenerics_1.14.0       matrixStats_1.1.0          </span></span>
<span><span class="co">#&gt; [13] ExperimentHub_2.10.0        AnnotationHub_3.10.0       </span></span>
<span><span class="co">#&gt; [15] BiocFileCache_2.10.1        dbplyr_2.4.0               </span></span>
<span><span class="co">#&gt; [17] BiocGenerics_0.48.1         escheR_1.3.1               </span></span>
<span><span class="co">#&gt; [19] ggplot2_3.4.4               BiocStyle_2.30.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.1.3                     bitops_1.0-7                 </span></span>
<span><span class="co">#&gt;   [3] gridExtra_2.3                 rlang_1.1.2                  </span></span>
<span><span class="co">#&gt;   [5] magrittr_2.0.3                compiler_4.3.2               </span></span>
<span><span class="co">#&gt;   [7] RSQLite_2.3.3                 png_0.1-8                    </span></span>
<span><span class="co">#&gt;   [9] systemfonts_1.0.5             vctrs_0.6.4                  </span></span>
<span><span class="co">#&gt;  [11] stringr_1.5.0                 pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt;  [13] crayon_1.5.2                  fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [15] backports_1.4.1               magick_2.8.1                 </span></span>
<span><span class="co">#&gt;  [17] XVector_0.42.0                ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt;  [19] labeling_0.4.3                utf8_1.2.4                   </span></span>
<span><span class="co">#&gt;  [21] promises_1.2.1                rmarkdown_2.25               </span></span>
<span><span class="co">#&gt;  [23] ragg_1.2.6                    purrr_1.0.2                  </span></span>
<span><span class="co">#&gt;  [25] bit_4.0.5                     xfun_0.41                    </span></span>
<span><span class="co">#&gt;  [27] zlibbioc_1.48.0               cachem_1.0.8                 </span></span>
<span><span class="co">#&gt;  [29] jsonlite_1.8.7                blob_1.2.4                   </span></span>
<span><span class="co">#&gt;  [31] highr_0.10                    later_1.3.1                  </span></span>
<span><span class="co">#&gt;  [33] DelayedArray_0.28.0           interactiveDisplayBase_1.40.0</span></span>
<span><span class="co">#&gt;  [35] broom_1.0.5                   R6_2.5.1                     </span></span>
<span><span class="co">#&gt;  [37] RColorBrewer_1.1-3            bslib_0.5.1                  </span></span>
<span><span class="co">#&gt;  [39] stringi_1.7.12                car_3.1-2                    </span></span>
<span><span class="co">#&gt;  [41] jquerylib_0.1.4               Rcpp_1.0.11                  </span></span>
<span><span class="co">#&gt;  [43] bookdown_0.36                 knitr_1.45                   </span></span>
<span><span class="co">#&gt;  [45] httpuv_1.6.12                 Matrix_1.6-2                 </span></span>
<span><span class="co">#&gt;  [47] tidyselect_1.2.0              abind_1.4-5                  </span></span>
<span><span class="co">#&gt;  [49] yaml_2.3.7                    curl_5.1.0                   </span></span>
<span><span class="co">#&gt;  [51] lattice_0.22-5                tibble_3.2.1                 </span></span>
<span><span class="co">#&gt;  [53] shiny_1.7.5.1                 withr_2.5.2                  </span></span>
<span><span class="co">#&gt;  [55] KEGGREST_1.42.0               evaluate_0.23                </span></span>
<span><span class="co">#&gt;  [57] desc_1.4.2                    Biostrings_2.70.1            </span></span>
<span><span class="co">#&gt;  [59] pillar_1.9.0                  BiocManager_1.30.22          </span></span>
<span><span class="co">#&gt;  [61] filelock_1.0.2                carData_3.0-5                </span></span>
<span><span class="co">#&gt;  [63] generics_0.1.3                rprojroot_2.0.4              </span></span>
<span><span class="co">#&gt;  [65] RCurl_1.98-1.13               BiocVersion_3.18.0           </span></span>
<span><span class="co">#&gt;  [67] munsell_0.5.0                 scales_1.2.1                 </span></span>
<span><span class="co">#&gt;  [69] xtable_1.8-4                  glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [71] tools_4.3.2                   ggsignif_0.6.4               </span></span>
<span><span class="co">#&gt;  [73] fs_1.6.3                      cowplot_1.1.1                </span></span>
<span><span class="co">#&gt;  [75] grid_4.3.2                    tidyr_1.3.0                  </span></span>
<span><span class="co">#&gt;  [77] AnnotationDbi_1.64.1          colorspace_2.1-0             </span></span>
<span><span class="co">#&gt;  [79] GenomeInfoDbData_1.2.11       cli_3.6.1                    </span></span>
<span><span class="co">#&gt;  [81] rappdirs_0.3.3                textshaping_0.3.7            </span></span>
<span><span class="co">#&gt;  [83] fansi_1.0.5                   S4Arrays_1.2.0               </span></span>
<span><span class="co">#&gt;  [85] viridisLite_0.4.2             dplyr_1.1.3                  </span></span>
<span><span class="co">#&gt;  [87] gtable_0.3.4                  rstatix_0.7.2                </span></span>
<span><span class="co">#&gt;  [89] sass_0.4.7                    digest_0.6.33                </span></span>
<span><span class="co">#&gt;  [91] SparseArray_1.2.2             farver_2.1.1                 </span></span>
<span><span class="co">#&gt;  [93] rjson_0.2.21                  memoise_2.0.1                </span></span>
<span><span class="co">#&gt;  [95] htmltools_0.5.7               pkgdown_2.0.7                </span></span>
<span><span class="co">#&gt;  [97] lifecycle_1.0.4               httr_1.4.7                   </span></span>
<span><span class="co">#&gt;  [99] mime_0.12                     bit64_4.0.5</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Boyi Guo, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
